lib/puppet/pops/types/type_calculator.rb

```diff
@@ -452,9 +452,18 @@ def instance_of_PTupleType(t, o)
 
   def instance_of_PStructType(t, o)
     return false unless o.is_a?(Hash)
-    h = t.hashed_elements
```

Comment:
Because a Struct may well be very short-lived (a Hash is inferred to a Struct) and the cache in that case causes more harm than good. Reverse the question, why use it when it's not needed here?